src/OpenTelemetry.Api/Trace/SpanAttributes.cs

```diff
@@ -0,0 +1,118 @@
+﻿// <copyright file="SpanAttributes.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// A class that represents the span attributes.
+    /// </summary>
+    public struct SpanAttributes
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SpanAttributes"/> struct.
+        /// </summary>
+        /// <param name="initialAttributes">Initial attributes for the <see cref="SpanAttributes"/>.</param>
+        public SpanAttributes(IEnumerable<KeyValuePair<string, object>> initialAttributes)
+        {
+            this.Attributes = initialAttributes == null ? new ActivityTagsCollection() : new ActivityTagsCollection(initialAttributes);
+        }
+
+        internal ActivityTagsCollection Attributes { get; }
```

Comment:
the getter should check if Attributes is null, and initialize.

Reply:
Since we always pass by the constructor,we won't have that possibility

Reply:
I see.
var spanAttribute = new SpanAttributes(); // does this line compile?

Reply:
no, because i didn't add the default constructor because struct does not allow it.